test/tests/functional/pbs_standing_resv_end_hook.py

```diff
@@ -0,0 +1,172 @@
+from tests.functional import *
+import time
+
+
+class TestStandingReservationEndHook(TestFunctional):
+    """
+    Tests to verify the reservation end hook for
+    a standing reservation if the occurrences end or get deleted
+    """
+
+    def setUp(self):
+        """ Create and import the python script for reservation end hook
+            in setup method"""
+
+        super(TestStandingReservationEndHook, self).setUp()
+        hook_script = """
+import pbs
+e=pbs.event()
+
+pbs.logmsg(pbs.LOG_DEBUG, 'Reservation End Hook name - %s' % (pbs.event().hook_name))
+
+if e.type == pbs.RESV_END:
+    pbs.logmsg(pbs.LOG_DEBUG, 'Reservation occurrence - %s'%pbs.event().resv.reserve_index)
+"""
+        self.hook_name = 'resvend_hook'
+        attrs = {'event': 'resv_end'}
+        self.server.create_import_hook(
+            self.hook_name, attrs, hook_script, overwrite=True)
+
+        self.server.manager(
+            MGR_CMD_SET, SERVER, {
+                'log_events': 2047}, expect=True)
+
+    def submit_and_confirm_resv(
+            self,
+            offset,
+            duration,
+            select="1:ncpus=1",
+            rrule='FREQ=MINUTELY;COUNT=2'):
+        """ Method to submit and confirm the standing reservation """
+
+        if 'PBS_TZID' in self.conf:
+            tzone = self.conf['PBS_TZID']
+        elif 'PBS_TZID' in os.environ:
+            tzone = os.environ['PBS_TZID']
+        else:
+            self.logger.info('Missing timezone, using Asia/Kolkata')
+            tzone = 'Asia/Kolkata'
+
+        start_time = time.time() + offset
+        end_time = start_time + duration
+
+        attrs = {
+            'reserve_start': start_time,
+            'reserve_end': end_time,
+            'Resource_List.select': select,
+            ATTR_resv_rrule: rrule,
+            ATTR_resv_timezone: tzone}
+
+        rid = self.server.submit(Reservation(TEST_USER, attrs))
+
+        resv_attrs = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        self.server.expect(RESV, resv_attrs, id=rid)
+
+        return rid
+
+    def test_delete_resv(self):
+        """ Test to delete one of the occurrences of a confirmed
+            standing reservation and verify the resvend hook. """
+
+        offset = 3
+        duration = 10
+        rid = self.submit_and_confirm_resv(offset, duration)
+
+        time.sleep(10)
+
+        msg = "Hook;Server@" + self.server.hostname.lower() + \
+            ";Reservation occurrence - %s" % '1'
```

Comment:
1 is a string literal, why not put it directly into the string instead of using string formatting. (and for other occurrences in the test plan)